pkg/cmd/cli/cmd/newapp.go

```diff
@@ -543,7 +542,7 @@ func setAnnotations(annotations map[string]string, result *newcmd.AppResult) err
 	for _, object := range result.List.Items {
 		err := util.AddObjectAnnotations(object, annotations)
 		if err != nil {
-			return err
+			return fmt.Errorf("failed to add annotation to object of type %q, this type is probably unknown.", object.GetObjectKind().GroupVersionKind())
```

Comment:
Honestly i think that's just more confusing for users.  There's nothing they can do to make the object implement the necessary interface, so that's not going to help them.

What i'm trying to convey here is "your openshift version doesn't understand this resource type/it's not a valid resource type"